etna/transforms/embeddings/models/ts2vec.py

```diff
@@ -0,0 +1,287 @@
+        sliding_length:
+            The length of sliding window. When this param is specified, a sliding inference would be applied on the time series.
+        sliding_padding:
+            This param specifies the contextual data length used for inference every sliding windows.
+
+        Returns
+        -------
+        :
+            array with embeddings of shape (n_timestamps, n_segments * output_dim)
+
+        Notes
+        -----
+        Model works with the index sorted in the alphabetic order. Thus, output embeddings correspond to the segments,
+        sorted in alphabetic order.
+        """
+        last_timestamp = max(np.where(~df.isna().all(axis=1))[0])
+        df = df[: last_timestamp + 1]
+        x = self._prepare_data(df=df)
+        embeddings = self.embedding_model.encode(
+            data=x,
+            mask=mask,
+            encoding_window="full_series",
+            causal=False,
+            sliding_length=sliding_length,
+            sliding_padding=sliding_padding,
+            batch_size=self.batch_size,
+        )
+
+        n_timestamps = len(df.index)
+        embeddings = np.repeat(embeddings[np.newaxis, :, :], n_timestamps, axis=0).reshape(n_timestamps, -1)
+        return embeddings
+
+    def encode_window(
+        self,
+        df: pd.DataFrame,
+        mask: Union[
+            Literal["binomial"], Literal["continuous"], Literal["all_true"], Literal["all_false"], Literal["mask_last"]
+        ] = "all_true",
+        encoding_window: Optional[Union[Literal["multiscale"], int]] = None,
+        sliding_length: Optional[int] = None,
+        sliding_padding: int = 0,
+    ) -> np.ndarray:
+        """Create embeddings of each series timestamp.
+
+        Parameters
+        ----------
+        df:
+            Dataframe with data.
+        mask:
+            The mask used by encoder on the test phase can be specified with this parameter. The possible options are:
+                * 'binomial' - mask timestamp with probability 0.5 (default one, used in the paper). It is used on the training phase.
+                * 'continuous' - mask random windows of timestamps
+                * 'all_true' - mask none of the timestamps
+                * 'all_false' - mask all timestamps
+                * 'mask_last' - mask last timestamp
+        sliding_length:
+            The length of sliding window. When this param is specified, a sliding inference would be applied on the time series.
+        sliding_padding:
+            The contextual data length used for inference every sliding windows.
+        encoding_window:
+            When this param is specified, the computed representation would the max pooling over this window. The possible options are:
+                * 'multiscale'
+                * integer specifying the pooling kernel size.
+
+        Returns
+        -------
+        :
+            array with embeddings of shape (n_timestamps, n_segments * output_dim)
```

Comment:
Why can't we return the result in shape (n_timestamps, n_segments, output_dim)? It seems like separation of n_segments and output_dim makes the result more clear and fixes the problem with ordering of the segments.